docs/design-documents/20241024-conduit-cli.md

```diff
@@ -0,0 +1,410 @@
+
+#### `--help`
+
+```bash
+conduit pipelines init
+conduit pipelines init my-first-pipeline
+conduit pipelines init my-first-pipeline --pipelines.path ~/my-other-path
+conduit pipelines init --source file@v1.0 --destination file
+```
+
+### `conduit pipelines ls`
+
+This command will list the running pipelines.
+
+It requires having Conduit previously running.
+
+#### Flags
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+
+#### `--help`
+
+```bash
+$ conduit pipelines ls
+NAME            STATUS
+my-pipeline     running
+my-other        degraded      
+```
+
+### `conduit pipelines describe ID`
+
+This command will describe the topology of the pipeline.
+
+It requires having Conduit previously running.
+
+It requires the pipeline id as argument.
+
+#### Arguments
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| id  |  pipeline id to describe | Yes | |
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+
+#### `--help`
+
+```bash
+$ conduit pipelines describe [ID]
+
+EXAMPLE:
+
+$ conduit pipelines describe my-pipeline
+ID: generator-to-log
+State:
+  Status: STATUS_STOPPED
+Config:
+  Name: generator-to-log
+  Description: Postgres source, file destination
+Connector IDs:
+  - generator-to-log:postgres-source
+  - generator-to-log:destination
+Processor IDs:
+  - my-processor:avro.encode
+Created At: 2024-10-09T13:55:17.113068Z
+Updated At: 2024-10-09T13:55:17.113802Z
+```
+
+### `conduit connectors ls`
+
+This command will list all the available connectors.
+
+It requires having Conduit previously running.
+
+#### `--help`
+
+```bash
+$ conduit connectors ls
+ID                PLUGIN                          TYPE         PIPELINE
+my-source         postgres@v0.2.0                 builtin      file-to-postgres
+my-destination    conduit-connector-http@0.1.0.   standalone   my-other-pipeline
+```
+
+### `conduit connectors describe ID`
+
+This command will describe the existing running connector.
+
+It requires having Conduit previously running.
+
+#### Flags
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+
+#### `--help`
+
+```bash
+$ conduit connnectors describe ID [--grpc.address]
+
+EXAMPLE:
+
+$ conduit connectors describe my-source
+NAME        URL             PIPELINE        PLUGIN
+my-source   https://...     my-pipeline     postgres@v0.2.0
+```
+
+### `conduit connector-plugins ls`
+
+This command will list all the available connector plugins.
+
+It does not require having Conduit previously running.
+
+#### `--help`
+
+```bash
+$ conduit connector-plugins ls
+PLUGIN                          
+postgres@v0.2.0                 
+conduit-connector-http@0.1.0
+```
+
+### `conduit connector-plugins PLUGIN`
```

Comment:
Shouldn't this then be `conduit connector-plugins describe PLUGIN`?

Reply:
@lovromazgon yeah, just fixed